scripts/helpers/buildExtensionManifest.js

```diff
@@ -0,0 +1,963 @@
+    },
+    com_adobe_identity: {
+      type: "object",
+      properties: {
+        idSyncContainerId: {
+          type: "integer"
+        }
+      },
+      required: ["idSyncContainerId"]
+    },
+    com_adobe_target: {
+      type: "object",
+      properties: {
+        propertyToken: {
+          type: "string"
+        }
+      },
+      required: ["propertyToken"]
+    }
+  };
+  const configOverridesWithDatastream = {
+    ...configOverridesProps,
+    sandbox: {
+      type: "string",
+      minLength: 1
+    },
+    datastreamId: {
+      type: "string",
+      minLength: 1
+    },
+    datastreamIdInputMethod: {
+      type: "string",
+      enum: ["freeform", "select"]
+    }
+  };
+  return {
+    type: "object",
+    additionalProperties: false,
+    properties: {
+      ...configOverridesProps,
+      development: {
+        type: "object",
+        additionalProperties: false,
+        properties: {
+          ...(isAction ? configOverridesWithDatastream : configOverridesProps)
+        }
+      },
+      staging: {
+        type: "object",
+        additionalProperties: false,
+        properties: {
+          ...(isAction ? configOverridesWithDatastream : configOverridesProps)
+        }
+      },
+      production: {
+        type: "object",
+        additionalProperties: false,
+        properties: {
+          ...(isAction ? configOverridesWithDatastream : configOverridesProps)
+        }
+      }
+    }
+  };
+};
+
+/**
+ * Create a list of common transforms used to with edge config overrides.
+ * Works for both actions and extension configuration.
+ * @param {boolean} isAction
+ * @returns {{ type: "remove", propertyPath: string }[]}
+ */
+const createEdgeConfigOverridesTransforms = isAction => {
+  const prefix = isAction ? "" : "instances[].";
+  return [
+    {
+      type: "remove",
+      propertyPath: `${prefix}edgeConfigOverrides.development.sandbox`
+    },
+    {
+      type: "remove",
+      propertyPath: `${prefix}edgeConfigOverrides.staging.sandbox`
+    },
+    {
+      type: "remove",
+      propertyPath: `${prefix}edgeConfigOverrides.production.sandbox`
+    },
+    {
+      type: "remove",
+      propertyPath: `${prefix}edgeConfigOverrides.development.datastreamIdInputMethod`
+    },
+    {
+      type: "remove",
+      propertyPath: `${prefix}edgeConfigOverrides.staging.datastreamIdInputMethod`
+    },
+    {
+      type: "remove",
+      propertyPath: `${prefix}edgeConfigOverrides.production.datastreamIdInputMethod`
+    }
+  ];
+};
+
+/**
+ * Create the contents of the extension.json aka the extension definition.
+ *
+ * @param {ExtensionManifestConfiguration} options
+ * @returns {ExtensionManifest}
+ */
+const createExtensionManifest = ({ version }) => {
+  const actionEdgeConfigOverridesSchema = createEdgeConfigOverridesSchema(true);
+  const actionEdgeConfigOverridesTransforms = createEdgeConfigOverridesTransforms(
+    true
+  );
+  /** @type {ExtensionManifest} */
```

Comment:
This is the section that will most often change. It may be helpful to have this in its own file to make it easier to modify. i.e. you could have a file "createExtensionManifest" that accepts { version, actionEdgeConfigOverridesSchema, actionEdgeConfigOverridesTrasforms }